cmd/hstspreload/main.go

```diff
@@ -0,0 +1,203 @@
+  hstspreload -h "max-age=10886400; includeSubDomains"
+
+Return code:
+
+  0    Passed all checks.
+  1    Error (failed at least one requirement).
+  2    Had warnings, but passed all requirements.
+  3    Invalid commandline arguments
+  4    Displayed help
+
+`)
+		os.Exit(4)
+		return
+	}
+
+	var header *string
+	var issues hstspreload.Issues
+
+	switch args[0] {
+	case "+h", "preloadableheader":
+		issues = preloadableHeader(args[1])
+
+	case "-h", "removableheader":
+		issues = removableHeader(args[1])
+
+	case "+d", "preloadabledomain":
+		header, issues = preloadableDomain(args[1])
+
+	case "-d", "removabledomain":
+		header, issues = removableDomain(args[1])
+
+	case "status":
+		l, err := chromiumpreload.GetLatest()
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "%s\n", err)
+		}
+		m := chromiumpreload.PreloadEntriesToMap(l)
+		fmt.Printf("Status: %v", m[chromiumpreload.Domain(args[1])])
+		os.Exit(0)
+
+	default:
+		fmt.Printf("Unknown command: %s\n", args[0])
+		os.Exit(3)
+	}
+
+	// Wrap this in a function to (statically) enforce a return code.
+	showResult := func() int {
+		if header != nil {
+			fmt.Printf("Observed header: %s%s%s\n", bold, *header, reset)
+		}
+
+		fmt.Printf("\n")
+		switch {
+		case len(issues.Errors) > 0:
+			return 1
+
+		case len(issues.Warnings) > 0:
+			return 2
+
+		default:
+			fmt.Printf("%sSatisfies requirements.%s\n\n", green, reset)
+			return 0
+		}
+	}
+	exitCode := showResult()
+
+	printList(issues.Errors, "Error", red)
+	printList(issues.Warnings, "Warning", yellow)
+
+	os.Exit(exitCode)
+}
+
+func preloadableHeader(header string) (issues hstspreload.Issues) {
+	warnIfNotHeader(header)
+
+	fmt.Printf(
+		"Checking header \"%s%s%s\" for preload requirements...\n",
+		bold, header, reset)
+
+	return hstspreload.PreloadableHeaderString(header)
+}
+
+func removableHeader(header string) (issues hstspreload.Issues) {
+	warnIfNotHeader(header)
+
+	fmt.Printf(
+		"Checking header \"%s%s%s\" for removal requirements...\n",
+		bold, header, reset)
+
+	return hstspreload.RemovableHeaderString(header)
+}
+
+func preloadableDomain(domain string) (header *string, issues hstspreload.Issues) {
+	mustBeDomain(domain)
+
+	fmt.Printf(
+		"Checking domain %s%s%s for preload requirements...\n",
+		underline, domain, reset)
+
+	return hstspreload.PreloadableDomain(domain)
+}
+
+func removableDomain(domain string) (header *string, issues hstspreload.Issues) {
+	mustBeDomain(domain)
+
+	fmt.Printf(
+		"Checking domain %s%s%s for removal requirements...\n",
+		underline, domain, reset)
+
+	return hstspreload.RemovableDomain(domain)
+}
+
+func warnIfNotHeader(str string) {
+	if probablyURL(str) {
+		fmt.Fprintf(os.Stderr,
+			"Warning: please supply an HSTS header string (it appears you supplied a URL).\n")
+	}
+	if probablyDomain(str) {
+		fmt.Fprintf(os.Stderr,
+			"Warning: please supply an HSTS header string (it appears you supplied a domain).\n")
+	}
+}
+
+func mustBeDomain(str string) {
```

Comment:
Sounds like this does a similar job as warnIfNotHeader. You should use the same naming scheme for both.

(warnIfNotDomain for this, or mustBeHeader for the previous function)